src/test/lib/ApiTest.cpp

```diff
@@ -1753,8 +1753,210 @@ class SecConfigTestContext {
     }
 };
 
+void SettingApplyTests(HQUIC Handle, uint32_t Param) {
+    {
+        QUIC_SETTINGS Settings{0};
+        Settings.MaxAckDelayMs = UINT32_MAX;
```

Comment:
Along with all these negative `UINT32_MAX` size tests, it's probably a good idea to add tests that verify that `QUIC_VAR_INT_MAX` does work.